lib/Doctrine/ORM/Internal/Hydration/ObjectHydrator.php

```diff
@@ -148,6 +151,12 @@ protected function hydrateAllData()
             $coll->takeSnapshot();
         }
 
+        foreach ($this->uninitializedCollections as $coll) {
+            if (! $coll->isInitialized()) {
```

Comment:
> the collection will be considered "existed", and "existed" collections are only replaced if REFRESH hint is present

Did you mean to write "initialized"?

Reply:
No, there is such a logic that if a Collection is PersistentCollection and initialized it is added to "existing"
```
else {
            // Is already PersistentCollection, and DON'T REFRESH or FETCH-JOIN!
            $this->existingCollections[$oid . $fieldName] = $value;
 ```

Reply:
I see 👍; please use "existing" then.